### PR TITLE
chore: bump `indexmap` and use `edition=2021`

### PR DESCRIPTION
**Pull request by attila-lin**
oh, failed with no std

### CODE REVIEW

File: src/priority_queue/iterators.rs [hunk 1]
```diff
@@ -105,6 +111,7 @@ impl<'a, I: 'a, P: 'a, H: 'a> Drop for IterMut<'a, I, P, H>
 where
     I: Hash + Eq,
     P: Ord,
+    H: BuildHasher,
```
Comment (garro95) [inline src/priority_queue/iterators.rs:114]: Is this change needed?

Comment (attila-lin) [reply src/priority_queue/iterators.rs:114]: I think so.

Needed by `MutableKeys` trait
